crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -60,8 +60,9 @@ pub(crate) fn path_to_module(db: &dyn Db, path: &FilePath) -> Option<Module> {
 /// Resolves the module for the file with the given id.
 ///
 /// Returns `None` if the file is not a module locatable via any of the known search paths.
+#[allow(unreachable_pub)]
```

Comment:
I would prefer to fix the lint warning here instead, or to not make the type public (either way is fine)

Reply:
This is a bug with making Salsa tracked functions `pub`. Even if the tracked `pub` function is both reachable and used (which it is in this PR), Salsa also creates a struct by the same name, which is not reachable, so whenever you make a Salsa tracked function `pub` and expose it with `pub use`, you get this spurious lint warning. I don't know any way to fix it, unless we make this whole submodule `pub`, or move the function out to `lib.rs`.

Reply:
Ahh, a TODO would be helpful then.

I know how to fix it, upgrade salsa :laughing: